ethcore/builtin/src/lib.rs

```diff
@@ -44,6 +45,19 @@ trait Pricer: Send + Sync {
 	fn cost(&self, input: &[u8], at: u64) -> U256;
 }
 
+/// A linear pricing model. This computes a price using a base cost and a cost per-word.
+pub type Fixed = u64;
+
+impl Pricer for Fixed {
+	fn cost(&self, input: &[u8], _at: u64) -> U256 {
+		use std::convert::TryInto;
+		let (rounds_bytes, _) = input.split_at(std::mem::size_of::<u32>());
+		// todo[dvdplm] what's the error handling strategy here? panic? return U256::MAX? Is the input len checked before this call?
```

Comment:
`ModexpPricer` seem to return zero or `U256::max_value()`. Returning zero seems odd though: they get to run the precompile for free?
/cc @sorpaas

Reply:
@dvdplm if we return `U256::max_value()`, then the error on empty input would be out of gas, but the spec requires otherwise
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-152.md#test-vector-0
`execute` will do the length check and return the appropriate error.